src/reader/reader.ts

```diff
@@ -47,24 +49,153 @@ export function putContentIntoLines(contents: string): Query[] {
+  str = str.replace(/\]/g, "\\]");
+  str = str.replace(/\{/g, "\\{");
+  str = str.replace(/\}/g, "\\}");
+  str = str.replace(/\|/g, "\\|");
+  return str;
+}
+
+/**
+ * Strips content between two delimeters, starting with a regex
+ * Works across multiple lines (hence the name!) but preserves the newlines
+ * themselves to allow error reporting to reflect source lines.
+ *
+ * Note this requires dotAll regex (s) to work across multiple lines so only
+ * really works on node 9+.
+ */
+function multiLineReplace(
+  content: string,
+  start_regex: string,
+  start_delim: string,
+  end_delim: string
+): string {
+  // Build up a regex capturing both any start pattern,
+  // and the matched between the two delimeters
+  const regex = new RegExp(
+    "(" +
+      start_regex +
+      ")" +
+      "(" +
+      regexEscape(start_delim) +
+      ".*?" +
+      regexEscape(end_delim) +
+      ")",
+    "si"
+  );
+
+  // Preserve new lines but delete anything else between the separators
+  // You should get two matches
+  const match = regex.exec(content) || [];
+  if (match.length >= 2) {
+    // Replace everything between the delimters except newlines
+    // Note .* will not match newline without /s flag
+    const newlines = match[2].replace(/.*/g, "");
+    // Now replace whole regex with just the newlines left over
+    content = content.replace(regex, "$1" + newlines);
+  }
+  return content;
+}
+
+/**
+ * Strip comments as may contain SQL-like syntax which can confuse the lexer
+ * Preserve new lines so output still makes sense
  */
 function stripComments(content: string): string {
-  const contentInLines = content.split(Keyword.Newline);
-
-  for (let i = 0; i < contentInLines.length; i++) {
-    if (
-      contentInLines[i].startsWith(Keyword.CommentDash) ||
-      contentInLines[i].startsWith(Keyword.CommentHash) ||
-      contentInLines[i].startsWith(Keyword.CommentMultiLineStart)
-    ) {
-      contentInLines[i] = "";
-    }
+  // Add a new line to end in case not there to make regex matching easier
+  let added_new_line = false;
+  if (content[-1] != "\n") {
+    content = content + "\n";
+    added_new_line = true;
+  }
+
+  // Remove any content after a dash comment until end of line
+  const comment_dash_regex = new RegExp(Keyword.CommentDash + ".*?\n", "");
+  content = content.replace(comment_dash_regex, "\n");
+
+  // Remove any content after a hash comment until end of line
+  const comment_hash_regex = new RegExp(Keyword.CommentHash + ".*?\n", "");
+  content = content.replace(comment_hash_regex, "\n");
+
+  // Remove any content in a multi-line comment - but preserve new lines
+  content = multiLineReplace(
+    content,
+    "",
+    Keyword.CommentMultiLineStart,
+    Keyword.CommentMultiLineEnd
+  );
+
+  // Remove any new line we added
+  if (added_new_line) {
+    content = content.slice(0, -1);
+  }
+
+  return content;
+}
+
+/**
+ * Strip out anything of the format:
+ *   language js as $$ ... $$
```

Comment:
Potenitally newbie question, what dialect of SQL is this? I've never seen `language as` before.
I'm asking because I'm going to eventually split the codebase into drivers i.e. `mysql`, `postgres` etc...

Reply:
We use it in [BigQuery](https://cloud.google.com/bigquery/docs/reference/standard-sql/user-defined-functions) to run JavaScript. Not sure what @jirutka was using it for, for SQL.

Looks like [MySQL](https://dev.mysql.com/doc/refman/8.0/en/create-procedure.html) and [PostGres](https://www.postgresql.org/docs/9.1/sql-createfunction.html#SQL-CREATEFUNCTION-EXAMPLES) has a slight variant of it whereby you do it after the fact but only supports SQL.

Maybe this should be behind a switch?